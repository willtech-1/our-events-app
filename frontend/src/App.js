import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
// import components
import Signup from "./user/Signup";
import Signin from "./user/Signin";
import Home from "./components/Home";
import Navbar from "./components/Navbar";
import PrivateRoute from "./auth/PrivateRoute";
import AdminRoute from "./auth/AdminRoute";
import AddCategory from "./admin/AddCategory";
import AddEvent from "./admin/AddEvent";
import Events from "./components/Events";
import Footer from "./components/Footer"
import ManageEvents from "./admin/ManageEvents";
import UpdateEvent from "./admin/UpdateEvent";
import AdminDashboard from "./user/AdminDashboard";
import Dashboard from "./user/UserDashboard";
import Payment from "./components/Payment.js"

const App = () => {
  return (
    <BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/signup" exact element={<Signup />} />
        <Route path="/signin" exact element={<Signin />} />
        
         {/* user protected routes */}
        <Route element={<PrivateRoute />}>
          <Route path='/user/dashboard' exact element={<Dashboard />} /> 
          <Route path="/" exact element={<Home />} />
          <Route path="/events" exact element={<Events />} />
          <Route path="/payment" exact element={<Payment />} />
        </Route>
        
        {/* admin protected routes */}
        <Route element={<AdminRoute />}>
          <Route path='/admin/dashboard' exact element={<AdminDashboard />} />
          <Route path='/create/category' exact element={<AddCategory />} />
          <Route path='/create/event' exact element={<AddEvent />} />
          <Route path='/admin/events' exact element={<ManageEvents />} />
          <Route path='/admin/event/update/:eventId' exact element={<UpdateEvent />} />
          
        </Route>
      </Routes>
      <Footer />
    </BrowserRouter>
  );
};

export default App;
