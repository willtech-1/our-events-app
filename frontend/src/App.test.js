import React from "react"
import App from './App';
// import renderer
import renderer from 'react-test-renderer';

// describe what we are doing
describe('snapshot testing', () => {
  // test mode to create the test
  test("snapshot for App Component", () => {
    const component = renderer.create(<App />).toJSON()
    // toMatchSnapshot() matcher function of the expect object
    expect(component).toMatchSnapshot()
  })
})





