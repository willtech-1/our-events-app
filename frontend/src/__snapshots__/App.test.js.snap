// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for App Componen 1`] = `
Array [
  <div
    className="fixed w-full h-20 z-[100]"
    style={
      Object {
        "backgroundColor": "#F4F6F6",
      }
    }
  >
    <div
      className="flex justify-between items-center w-full h-full px-2 2xl:px-24"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="nav-logo"
        >
          <span
            className="span1"
          >
            Our
          </span>
          <span
            className="span2"
          >
            Events™
          </span>
        </div>
      </a>
      <div>
        <ul
          className="hidden md:flex"
        >
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Home
            </li>
          </a>
          <a
            href="/events"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Events
            </li>
          </a>
          <a
            href="/signup"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Signup
            </li>
          </a>
          <a
            href="/signin"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b border-white"
            >
              Signin
            </li>
          </a>
          <span
            onClick={[Function]}
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b border-white cursor-pointer"
            >
              Signout
            </li>
          </span>
        </ul>
        <div
          className="md:hidden"
          onClick={[Function]}
          style={
            Object {
              "color": "#1f2837",
            }
          }
        >
          <svg
            fill="currentColor"
            height={25}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width={25}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="fixed left-[-100%] top-0 p-10 ease-in duration-500"
      >
        <div>
          <div
            className="flex w-full items-center justify-between"
          >
            <a
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="nav-logo"
              >
                <span
                  className="span1"
                >
                  Our
                </span>
                <span
                  className="span2"
                >
                  Events™
                </span>
              </div>
            </a>
            <div
              className="rounded-full shadow-lg shadow-gray-400 p-3 cursor-pointer"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </div>
          </div>
          <div
            className="border-b border-gray-300 my-4"
          >
            <p
              className="w-[85%] md:w-[90%] py-4"
            >
              Let's come together.
            </p>
          </div>
        </div>
        <div
          className="py-4 flex flex-col"
        >
          <ul
            className="uppercase"
          >
            <a
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Home
              </li>
            </a>
            <a
              href="/events"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Events
              </li>
            </a>
            <a
              href="/signup"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Signup
              </li>
            </a>
            <a
              href="/signin"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm cursor-pointer"
                onClick={[Function]}
              >
                Signin
              </li>
            </a>
            <span
              onClick={[Function]}
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Signout
              </li>
            </span>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="bg-white dark:bg-gray-900"
  >
    <div
      className="py-6 px-4 bg-gray-100 dark:bg-gray-700 md:flex md:items-center md:justify-between"
    >
      <span
        className="text-sm text-gray-500 dark:text-gray-300 sm:text-center"
      >
        © 2022 
        <a
          href="https://flowbite.com/"
        >
          OurEvents™
        </a>
        . All Rights Reserved.
      </span>
      <div
        className="flex mt-4 space-x-6 sm:justify-center md:mt-0"
      >
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M22 12c0-5.523-4.477-10-10-10S2 6.477 2 12c0 4.991 3.657 9.128 8.438 9.878v-6.987h-2.54V12h2.54V9.797c0-2.506 1.492-3.89 3.777-3.89 1.094 0 2.238.195 2.238.195v2.46h-1.26c-1.243 0-1.63.771-1.63 1.562V12h2.773l-.443 2.89h-2.33v6.988C18.343 21.128 22 16.991 22 12z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            Facebook page
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.29 20.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0022 5.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.072 4.072 0 012.8 9.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 012 18.407a11.616 11.616 0 006.29 1.84"
            />
          </svg>
          <span
            className="sr-only"
          >
            Twitter page
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            GitHub account
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10c5.51 0 10-4.48 10-10S17.51 2 12 2zm6.605 4.61a8.502 8.502 0 011.93 5.314c-.281-.054-3.101-.629-5.943-.271-.065-.141-.12-.293-.184-.445a25.416 25.416 0 00-.564-1.236c3.145-1.28 4.577-3.124 4.761-3.362zM12 3.475c2.17 0 4.154.813 5.662 2.148-.152.216-1.443 1.941-4.48 3.08-1.399-2.57-2.95-4.675-3.189-5A8.687 8.687 0 0112 3.475zm-3.633.803a53.896 53.896 0 013.167 4.935c-3.992 1.063-7.517 1.04-7.896 1.04a8.581 8.581 0 014.729-5.975zM3.453 12.01v-.26c.37.01 4.512.065 8.775-1.215.25.477.477.965.694 1.453-.109.033-.228.065-.336.098-4.404 1.42-6.747 5.303-6.942 5.629a8.522 8.522 0 01-2.19-5.705zM12 20.547a8.482 8.482 0 01-5.239-1.8c.152-.315 1.888-3.656 6.703-5.337.022-.01.033-.01.054-.022a35.318 35.318 0 011.823 6.475 8.4 8.4 0 01-3.341.684zm4.761-1.465c-.086-.52-.542-3.015-1.659-6.084 2.679-.423 5.022.271 5.314.369a8.468 8.468 0 01-3.655 5.715z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            Dribbble account
          </span>
        </a>
      </div>
    </div>
  </footer>,
]
`;

exports[`snapshot testing snapshot for App Component 1`] = `
Array [
  <div
    className="fixed w-full h-20 z-[100]"
    style={
      Object {
        "backgroundColor": "#F4F6F6",
      }
    }
  >
    <div
      className="flex justify-between items-center w-full h-full px-2 2xl:px-24"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="nav-logo"
        >
          <span
            className="span1"
          >
            Our
          </span>
          <span
            className="span2"
          >
            Events™
          </span>
        </div>
      </a>
      <div>
        <ul
          className="hidden md:flex"
        >
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Home
            </li>
          </a>
          <a
            href="/events"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Events
            </li>
          </a>
          <a
            href="/signup"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b"
            >
              Signup
            </li>
          </a>
          <a
            href="/signin"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b border-white"
            >
              Signin
            </li>
          </a>
          <span
            onClick={[Function]}
          >
            <li
              className="ml-10 text-sm uppercase hover:border-b border-white cursor-pointer"
            >
              Signout
            </li>
          </span>
        </ul>
        <div
          className="md:hidden"
          onClick={[Function]}
          style={
            Object {
              "color": "#1f2837",
            }
          }
        >
          <svg
            fill="currentColor"
            height={25}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width={25}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="fixed left-[-100%] top-0 p-10 ease-in duration-500"
      >
        <div>
          <div
            className="flex w-full items-center justify-between"
          >
            <a
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="nav-logo"
              >
                <span
                  className="span1"
                >
                  Our
                </span>
                <span
                  className="span2"
                >
                  Events™
                </span>
              </div>
            </a>
            <div
              className="rounded-full shadow-lg shadow-gray-400 p-3 cursor-pointer"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </div>
          </div>
          <div
            className="border-b border-gray-300 my-4"
          >
            <p
              className="w-[85%] md:w-[90%] py-4"
            >
              Let's come together.
            </p>
          </div>
        </div>
        <div
          className="py-4 flex flex-col"
        >
          <ul
            className="uppercase"
          >
            <a
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Home
              </li>
            </a>
            <a
              href="/events"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Events
              </li>
            </a>
            <a
              href="/signup"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Signup
              </li>
            </a>
            <a
              href="/signin"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <li
                className="py-4 text-sm cursor-pointer"
                onClick={[Function]}
              >
                Signin
              </li>
            </a>
            <span
              onClick={[Function]}
            >
              <li
                className="py-4 text-sm"
                onClick={[Function]}
              >
                Signout
              </li>
            </span>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="bg-white dark:bg-gray-900"
  >
    <div
      className="py-6 px-4 bg-gray-100 dark:bg-gray-700 md:flex md:items-center md:justify-between"
    >
      <span
        className="text-sm text-gray-500 dark:text-gray-300 sm:text-center"
      >
        © 2022 
        <a
          href="https://flowbite.com/"
        >
          OurEvents™
        </a>
        . All Rights Reserved.
      </span>
      <div
        className="flex mt-4 space-x-6 sm:justify-center md:mt-0"
      >
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M22 12c0-5.523-4.477-10-10-10S2 6.477 2 12c0 4.991 3.657 9.128 8.438 9.878v-6.987h-2.54V12h2.54V9.797c0-2.506 1.492-3.89 3.777-3.89 1.094 0 2.238.195 2.238.195v2.46h-1.26c-1.243 0-1.63.771-1.63 1.562V12h2.773l-.443 2.89h-2.33v6.988C18.343 21.128 22 16.991 22 12z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            Facebook page
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.29 20.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0022 5.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.072 4.072 0 012.8 9.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 012 18.407a11.616 11.616 0 006.29 1.84"
            />
          </svg>
          <span
            className="sr-only"
          >
            Twitter page
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            GitHub account
          </span>
        </a>
        <a
          className="text-gray-400 hover:text-gray-900 dark:hover:text-white"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10c5.51 0 10-4.48 10-10S17.51 2 12 2zm6.605 4.61a8.502 8.502 0 011.93 5.314c-.281-.054-3.101-.629-5.943-.271-.065-.141-.12-.293-.184-.445a25.416 25.416 0 00-.564-1.236c3.145-1.28 4.577-3.124 4.761-3.362zM12 3.475c2.17 0 4.154.813 5.662 2.148-.152.216-1.443 1.941-4.48 3.08-1.399-2.57-2.95-4.675-3.189-5A8.687 8.687 0 0112 3.475zm-3.633.803a53.896 53.896 0 013.167 4.935c-3.992 1.063-7.517 1.04-7.896 1.04a8.581 8.581 0 014.729-5.975zM3.453 12.01v-.26c.37.01 4.512.065 8.775-1.215.25.477.477.965.694 1.453-.109.033-.228.065-.336.098-4.404 1.42-6.747 5.303-6.942 5.629a8.522 8.522 0 01-2.19-5.705zM12 20.547a8.482 8.482 0 01-5.239-1.8c.152-.315 1.888-3.656 6.703-5.337.022-.01.033-.01.054-.022a35.318 35.318 0 011.823 6.475 8.4 8.4 0 01-3.341.684zm4.761-1.465c-.086-.52-.542-3.015-1.659-6.084 2.679-.423 5.022.271 5.314.369a8.468 8.468 0 01-3.655 5.715z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="sr-only"
          >
            Dribbble account
          </span>
        </a>
      </div>
    </div>
  </footer>,
]
`;
